redash/handlers/data_sources.py

```diff
@@ -95,6 +93,9 @@ def post(self):
                                                          name=req['name'],
                                                          type=req['type'],
                                                          options=config)
+
+        models.db.session.commit()
```

Comment:
I don't know if there's any easy fix for the tests. I'm going to look over all the PUT/POST/DELETE handlers to make sure they all have `commit()` calls where appropriate.